### PR TITLE
Add additional labels to daemonsets/deployments.

### PR DESCRIPTION
**Pull request by naemono**
related #733

Would you be opposed to also adding these `additionalLabels` the deployments/daemonsets that create the pods?

### CODE REVIEW

**Comment by daichimukai** [comment]
Hi @naemono, Thank you for the PR.

I feel the name of the existing field `additionalLabels` is ambiguous, so I think there is a case to be made for adding these labels to the resource or not. To clarify the semantics of the field I would like to:

- change `additionalLabel` to `podLabels`. `podLabels` is only applied to the pod template,
- add another field `labels` which is applied to the resource itself.

What do you think?

**Comment by naemono** [comment]
> What do you think?

I think your suggestion for 2 different fields is reasonable, and I will adjust this PR to do just that @daichimukai

**Comment by naemono** [comment]
And here's the helm templating showing these working as expected @daichimukai 
```
❯ helm template topo . --set lvmd.podLabels.pod=label --set lvmd.labels.owner=label --set controller.podLabels.pod=label --set controller.labels.owner=label --set node.podLabels.pod=label --set node.labels.owner=label --set scheduler.podLabels.pod=label --set scheduler.labels.owner=label -s templates/lvmd/daemonset.yaml | yq -o=json | jq '{metaLabels: .metadata.labels, podtemplateLabels: .spec.template.metadata.labels}'
{
  "metaLabels": {
    "idx": "0",
    "helm.sh/chart": "topolvm-11.4.0",
    "app.kubernetes.io/name": "topolvm",
    "app.kubernetes.io/instance": "topo",
    "app.kubernetes.io/version": "0.20.0",
    "app.kubernetes.io/managed-by": "Helm",
    "owner": "label"
  },
  "podtemplateLabels": {
    "idx": "0",
    "app.kubernetes.io/component": "lvmd",
    "app.kubernetes.io/name": "topolvm",
    "app.kubernetes.io/instance": "topo",
    "pod": "label"
  }
}

❯ helm template topo . --set lvmd.podLabels.pod=label --set lvmd.labels.owner=label --set controller.podLabels.pod=label --set controller.labels.owner=label --set node.podLabels.pod=label --set node.labels.owner=label --set scheduler.podLabels.pod=label --set scheduler.labels.owner=label -s templates/controller/deployment.yaml | yq -o=json | jq '{metaLabels: .metadata.labels, podtemplateLabels: .spec.template.metadata.labels}'
{
  "metaLabels": {
    "helm.sh/chart": "topolvm-11.4.0",
    "app.kubernetes.io/name": "topolvm",
    "app.kubernetes.io/instance": "topo",
    "app.kubernetes.io/version": "0.20.0",
    "app.kubernetes.io/managed-by": "Helm",
    "owner": "label"
  },
  "podtemplateLabels": {
    "app.kubernetes.io/component": "controller",
    "app.kubernetes.io/name": "topolvm",
    "app.kubernetes.io/instance": "topo",
    "pod": "label"
  }
}

❯ helm template topo . --set lvmd.podLabels.pod=label --set lvmd.labels.owner=label --set controller.podLabels.pod=label --set controller.labels.owner=label --set node.podLabels.pod=label --set node.labels.owner=label --set scheduler.podLabels.pod=label --set scheduler.labels.owner=label -s templates/node/daemonset.yaml | yq -o=json | jq '{metaLabels: .metadata.labels, podtemplateLabels: .spec.template.metadata.labels}'
{
  "metaLabels": {
    "helm.sh/chart": "topolvm-11.4.0",
    "app.kubernetes.io/name": "topolvm",
    "app.kubernetes.io/instance": "topo",
    "app.kubernetes.io/version": "0.20.0",
    "app.kubernetes.io/managed-by": "Helm",
    "owner": "label"
  },
  "podtemplateLabels": {
    "app.kubernetes.io/component": "node",
    "app.kubernetes.io/name": "topolvm",
    "app.kubernetes.io/instance": "topo",
    "pod": "label"
  }
}

❯ helm template topo . --set lvmd.podLabels.pod=label --set lvmd.labels.owner=label --set controller.podLabels.pod=label --set controller.labels.owner=label --set node.podLabels.pod=label --set node.labels.owner=label --set scheduler.podLabels.pod=label --set scheduler.labels.owner=label -s templates/scheduler/daemonset.yaml | yq -o=json | jq '{metaLabels: .metadata.labels, podtemplateLabels: .spec.template.metadata.labels}'
{
  "metaLabels": {
    "helm.sh/chart": "topolvm-11.4.0",
    "app.kubernetes.io/name": "topolvm",
    "app.kubernetes.io/instance": "topo",
    "app.kubernetes.io/version": "0.20.0",
    "app.kubernetes.io/managed-by": "Helm",
    "owner": "label"
  },
  "podtemplateLabels": {
    "app.kubernetes.io/component": "scheduler",
    "app.kubernetes.io/name": "topolvm",
    "app.kubernetes.io/instance": "topo",
    "pod": "label"
  }
}

❯ helm template topo . --set lvmd.podLabels.pod=label --set lvmd.labels.owner=label --set controller.podLabels.pod=label --set controller.labels.owner=label --set node.podLabels.pod=label --set node.labels.owner=label --set scheduler.podLabels.pod=label --set scheduler.labels.owner=label --set scheduler.type=deployment -s templates/scheduler/deployment.yaml | yq -o=json | jq '{metaLabels: .metadata.labels, podtemplateLabels: .spec.template.metadata.labels}'
{
  "metaLabels": {
    "helm.sh/chart": "topolvm-11.4.0",
    "app.kubernetes.io/name": "topolvm",
    "app.kubernetes.io/instance": "topo",
    "app.kubernetes.io/version": "0.20.0",
    "app.kubernetes.io/managed-by": "Helm",
    "owner": "label"
  },
  "podtemplateLabels": {
    "app.kubernetes.io/component": "scheduler",
    "app.kubernetes.io/name": "topolvm",
    "app.kubernetes.io/instance": "topo",
    "pod": "label"
  }
}
```

File: charts/topolvm/README.md [hunk 1]
```diff
@@ -83,7 +83,8 @@ You need to configure kube-scheduler to use topolvm-scheduler extender by referr
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | cert-manager.enabled | bool | `false` | Install cert-manager together. # ref: https://cert-manager.io/docs/installation/kubernetes/#installing-with-helm |
-| controller.additionalLabels | object | `{}` | Additional labels to be set on the controller pod |
+| controller.podLabels | object | `{}` | Additional labels to be set on the controller pod |
+| controller.labels | object | `{}` | Additional labels to be added to the Deployment |
```
Comment (daichimukai) [inline charts/topolvm/README.md:87]: This file is automatically generated by the `helm-docs` command. Please follow this instruction.

https://github.com/topolvm/topolvm/blob/07779ec247087d395288515dd16f2065edf98d19/charts/topolvm/README.md#update-readme

**Review by daichimukai — CHANGES_REQUESTED** [review]

File: charts/topolvm/README.md (hunk 1 above)
Comment (naemono) [reply charts/topolvm/README.md]: My apologies, I missed that. 👍

**Review by daichimukai — APPROVED** [review]

**Review by llamerada-jp — APPROVED** [review]

**Pull request merged**